### PR TITLE
Added error handling for CloudFlare specific errors

### LINKED ISSUE #549: Add more informative error messages for Coinbase historic API
Look into unofficial HTTP codes and their associated messages (Coinbase has timed-out a few times and appears to use Cloudflare). Consider using a common error message formatter, as this may apply to other APIs too.

https://en.wikipedia.org/wiki/List_of_HTTP_status_codes

### PR DESCRIPTION
**Pull request by rjmcneill**
Relates to #549 

Added on the historic data and the retrieval of the Coinbase products.

List of CloudFlare error codes:
https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#CloudFlare

CloudFlare API responses:
https://api.cloudflare.com/#responses

### CODE REVIEW

File: src/assets/js/app.js [hunk 1]
```diff
@@ -421,6 +422,22 @@ export default function() {
         render();
     }
 
+    function handleHTTPErrors(error) {
+        var statusText;
+
+        // Range of CloudFlare specific error codes
+        if (error.status >= 520 && error.status <= 526) {
+            statusText = 'CloudFlare error -';
+            for (var i = 0; i < error.response.errors.length; i++) {
```
Comment (jdhodges431) [inline src/assets/js/app.js]: How about:

```
error.response.errors.forEach(function(_error) {
    statusText += ' ' + _error.message;
});
```

**Comment by jdhodges431** [comment]
:+1:

File: src/assets/js/app.js (hunk 1 above)
Comment (rjmcneill) [inline src/assets/js/app.js]: Agreed!
Keep forgetting about forEach!

File: src/assets/js/app.js [hunk 2]
```diff
@@ -419,6 +420,22 @@ export default function() {
         render();
     }
 
+    function handleHTTPErrors(error) {
```
Comment (djmiley) [inline src/assets/js/app.js]: More of a CloudFlare specific naming maybe, if this is all the method is handling?

Comment (djmiley) [inline src/assets/js/app.js]: Or, if not, make the method more general? I may have missed something here though :(

Comment (rjmcneill) [inline src/assets/js/app.js]: @djmiley it actually handles both.
Previously error handling was just 
```JavaScript
statusText = error.statusText || 'Unknown reason.';
```
This was fine for the normal error codes, but the CloudFlare specific errors are not returned in the same way.
So to handle this problem, I added this function which checks for the specific CloudFlare error codes (520 - 526) and handles them appropriately.
If an error code falls outside this range, the error is just handled the same way as the previous implemented

Comment (djmiley) [inline src/assets/js/app.js]: Right, OK, sorry my brain didn't quite take things in before.

**Pull request closed without merging**